src/nunit.analyzers/ValuesUsage/ValuesUsageAnalyzer.cs

```diff
@@ -83,6 +84,12 @@ private static void AnalyzeParameter(SymbolAnalysisContext symbolContext, INamed
                         continue;
                     }
 
+#if NETSTANDARD2_0_OR_GREATER
+                    if (argumentSyntax.IsKind(SyntaxKind.SuppressNullableWarningExpression))
+                    {
+                        continue;
+                    }
+#endif
```

Comment:
This is the wrong spot to put this test. This allows code like `public void Test([Values(null!)] int i)`.
It now even allows `public void Test([Values(2!)] string s)`.
The analyzer detects a type mismatch, but the operator suppresses all errors when the null forgiveness operator is used.

Reply:
Nice catch, thanks!